### PR TITLE
[3.3.5] Beast Mastery hunter talent Animal Handler

### LINKED ISSUE #15361: [3.3.5][Hunter] Animal Handler
http://wotlk.openwow.com/spell=34454

Not increase the power of pet attack 10%

4c420344fc4c8508b2142a5031d9bf8b16c0e7fd
3.3.5 branch

### PR DESCRIPTION
**Pull request by ghost**
The goal of this PR is to restore the correct Animal Handler talent AP values for a BM hunter.

**Animal Handler, Rank 1**
- increases your pet's attack power by 5% and
- increases the duration of your Master's Call effect by 3 sec.

**Animal Handler, Rank 2**
- increases your pet's attack power by 10% and
- increases the duration of your Master's Call effect by 6 sec.

Viewed in talent calculator: http://wotlk.openwow.com/talent#cVbhzTGRb

by **Rushor**, closes  #15361

### CODE REVIEW

**Comment by Ascathor** [comment]
Wouldn't a spellscript make more sense ? Just asking to be sure.

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Comment by tkrokli** [comment]
Label detail: how is [Comp-C++Script] related to `src/server/game/Entities/Unit/StatSystem.cpp` ?

**Comment by Aokromes** [comment]
If there is no reason to merge this, i will merge it on 24h.

File: src/server/game/Entities/Unit/StatSystem.cpp [hunk 1]
```diff
@@ -1327,6 +1327,11 @@ void Guardian::UpdateAttackPowerAndDamage(bool ranged)
             }
 
             bonusAP = owner->GetTotalAttackPowerValue(RANGED_ATTACK) * 0.22f * mod;
+            if (AuraEffect* aurEff = owner->GetAuraEffectOfRankedSpell(34453, EFFECT_1, owner->GetGUID())) // Animal Handler
+            {
+                AddPct(bonusAP, aurEff->GetAmount());
+                AddPct(val, aurEff->GetAmount());
```
Comment (joschiwald) [inline src/server/game/Entities/Unit/StatSystem.cpp]: why is the bonus applied twice?

Comment (tkrokli) [inline src/server/game/Entities/Unit/StatSystem.cpp]: Good question, need feedback from @Rushor on this one.

Comment (ariel-) [inline src/server/game/Entities/Unit/StatSystem.cpp]: val is base AP
bonusAP is, well, bonus AP

Total AP = (val + bonusAP), so it's not applying twice, it's distributing multiplication

Comment (tkrokli) [inline src/server/game/Entities/Unit/StatSystem.cpp]: Thanks for the explanation. Looking at that code again, I wonder if the sequence of those 2 lines should be swapped, but maybe the order is indifferent like when performing a regular addition?

**Pull request merged**